app/src/main/java/com/bignerdranch/android/geoquiz/MainActivity.kt

```diff
@@ -3,64 +3,144 @@ package com.bignerdranch.android.geoquiz
-            currentIndex = (currentIndex + 1) % QuestionsBank.getBankSize()
+            moveToNext()
+            checkQuestion()
             updateQuestion()
         }
     }
 
-    private fun setupViewBinding() {
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        val view = binding.root
-        setContentView(view)
+    private fun checkQuestion() {
+        if (quizViewModel.answers[quizViewModel.currentIndex]) {
+            disableButtons()
+        } else {
+            enableButtons()
+        }
     }
 
     private fun updateQuestion() {
-        val questionTextResId = QuestionsBank.getQuestionText(currentIndex)
+        val questionTextResId = QuestionsBank.getQuestionText(quizViewModel.currentIndex)
         binding.questionTextView.setText(questionTextResId)
     }
 
+    private fun setAnswer() {
+        quizViewModel.answers[quizViewModel.currentIndex] = true
+        quizViewModel.answersCounter++
+    }
+
     private fun checkAnswer(userAnswer: Boolean) {
-        val correctAnswer = QuestionsBank.getQuestionAnswer(currentIndex)
-        val messageResId = if (userAnswer == correctAnswer) {
-            R.string.correct_toast
+        val correctAnswer = QuestionsBank.getQuestionAnswer(quizViewModel.currentIndex)
+        val messageResId: String
+        if (userAnswer == correctAnswer) {
+            quizViewModel.score++
+            messageResId = "Correct!"
         } else {
-            R.string.incorrect_toast
+            messageResId = "Incorrect!"
         }
-        Toast.makeText(this, messageResId, Toast.LENGTH_SHORT)
-                .show()
+        Toast.makeText(this, messageResId, Toast.LENGTH_SHORT).show()
+        disableButtons()
+    }
+
+    private fun checkResult() {
+        if (quizViewModel.answersCounter == QuestionsBank.getBankSize()) {
+            val result = ((quizViewModel.score.toDouble() / quizViewModel.answersCounter) * 100)
+            var resultMessage = "Your punctuation is $result %"
+            Toast.makeText(this, resultMessage, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun moveToPrev() {
+        quizViewModel.currentIndex =
+            if (quizViewModel.currentIndex - 1 < 0) {
+                QuestionsBank.getBankSize() - 1
+            } else {
+                quizViewModel.currentIndex - 1
+            }
+    }
+
+    private fun moveToNext() {
+        quizViewModel.currentIndex = (quizViewModel.currentIndex + 1) % QuestionsBank.getBankSize()
+    }
+
+    private fun disableButtons() {
+        binding.trueButton.isClickable = false
+        binding.trueButton.alpha = 0.5f
+        binding.falseButton.isClickable = false
+        binding.falseButton.alpha = 0.5f
+    }
+
+    private fun enableButtons() {
+        binding.trueButton.isClickable = true
+        binding.trueButton.alpha = 1f
+        binding.falseButton.isClickable = true
+        binding.falseButton.alpha = 1f
     }
```

Comment:
check enabled and disabled button states

Reply:
Would it be a selector with the buttons states? This time I used function "isEnabled" instead.